Initial draft of the migration guide for version 7 (#220)
neo4j/docs-graphql: modules/ROOT/pages/migration/index.adoc

== Breaking changes

Here is a list of all the breaking changes from version 6.0.0 to 7.0.0.

> === `@unique`

Reviewer: ```suggestion
=== Removed the `@unique` directive
```

> `@unique` directive is no longer supported:

Reviewer: ```suggestion
The `@unique` directive is no longer supported:
```

do we have a suggestion what a user could use instead of `@unique` ?

[source, graphql, indent=0]
----
type Movie {
    title: String! @unique()
}
----

> === Non-list relationships

Reviewer: we have "non-list relationships", "single element relationships" and "1-to-1 relationships" in this section, referring to the same(?)

we should pick one ("1-to-1 relationships?)

Author: I think the more accurate is non-list relationships or single element relationships. As 1-1 refers only to a subset of the cases.

Yeah, my bad, I'll change them all to be the same

> === Non-list relationships

Reviewer: ```suggestion
=== Deprecated single element relationships
```

> Deprecate single element relationships:

Reviewer: ```suggestion
Single element relationships have been deprecated in favor of list relationships:
```

> 

Reviewer: ```suggestion
```

> [source, graphql, indent=0]
> ----

Reviewer: ```suggestion
```

there are several suggestions deleting lines, since i added the before/after table

> type Movie {
>     director: Person @relationship(type: "DIRECTED", direction: "IN")

Reviewer: ```suggestion
[cols="1,1"]
|===
|Before | Now
a|
[source, graphql, indent=0]
----
type Movie {
    director: Person @relationship(type: "DIRECTED", direction: "IN")
}
----
a|
[source, graphql, indent=0]
----
type Movie {
    director: [Person!]! @relationship(type: "DIRECTED", direction: "IN")
}
----
|===
```

> }
> ----

Reviewer: ```suggestion
```

> In favor of list relationships:

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> [source, graphql, indent=0]
> ----

Reviewer: ```suggestion
```

> type Movie {
>     director: [Person!]! @relationship(type: "DIRECTED", direction: "IN")
> }
> ----

Reviewer: ```suggestion
```

> 1-1 relationships cannot be reliably enforced, leading to a data inconsistent with the schema. For this reason, these have been removed in favor of the more accurate list relationships.

Reviewer: ```suggestion
Single element relationships cannot be reliably enforced, leading to a data inconsistent with the schema.
```

what does "enforced" mean? sounds like there is a technical limitation in graphql?

=== `overwrite`

> Argument `overwrite` in `connect` operations has been deprecated. This is part of the overarching changes to how `connect` operation work in 7.x

Reviewer: ```suggestion
The `overwrite` argument in `connect` operations has been deprecated.
This is part of the overarching changes to how `connect` operations work in 7.x.
```

> For updating a relationship, use the `update` operation

Reviewer: ```suggestion
Instead, use the `update` operation to update a relationship.
```

> === `connectOrCreate`

Reviewer: ```suggestion
=== Removed the `connectOrCreate` operation
```

> The `connectOrCreate` operation has been removed due to limitations on its featureset when compared to other operations.

Reviewer: ```suggestion
The `connectOrCreate` operation has been removed due to limitations on its feature set when compared to other operations.
```

> `connectOrCreate` relies on `MERGE` operations in Cypher, which did not allowed for `onCreate` operations to follow nested relationships.

Reviewer: ```suggestion
`connectOrCreate` relies on `MERGE` operations in Cypher, which did not allow for `onCreate` operations to follow nested relationships.
```
